### PR TITLE
Re-enable secrets for packages disabled due to Kibana versions below …

### PR DESCRIPTION
**Pull request by ritalwar**
## Proposed commit message

Restore the secret for packages that were disabled because of Kibana versions below 8.12, as they were causing errors.

Secrets were reverted for older stack version earlier in this PR - https://github.com/elastic/integrations/pull/9279 

Now it is enabled only for stack 8.12 onwards.

For Prometheus and CockroachDB it is enabled for version 8.12.1 onwards, due to mapping conflicts resolved in the Fleet plugin.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates [#1305](https://github.com/elastic/obs-infraobs-team/issues/1305)

### CODE REVIEW

**Review by aliabbas-elastic — APPROVED** [review]
@ritalwar I tried loading the integrations with default configs in `8.12.0` and 2 integrations (`cockroachdb` and `prometheus`) in `8.12.1` & I didn't face any errors loading the packages. LGTM for the changes !!

**Review by harnish-elastic — APPROVED** [review]

File: packages/cockroachdb/changelog.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+- version: "1.9.0"
+  changes:
+    - description: Enable secrets with Kibana upgrade to 8.12, resolving errors.
```
Comment (ishleenk17) [inline packages/cockroachdb/changelog.yml:3]: The version should be 8.12.1 in changelog here.

**Comment by ritalwar** [comment]
> @ritalwar, some packages are marked as: `version: "^8.12.0" `and some are marked as: `version: "^8.12.1"`, what is the difference here?

In version 8.12.0 of Kibana, the Fleet plugin was refactored, fixing how dynamic mappings for integrations are generated. However, this change caused mapping conflicts during data ingestion, impacting integrations like Prometheus and CockroachDB. The issue was resolved in version 8.12.1, therefore an upgrade to this version for these packages. Detailed information [here](https://support.elastic.dev/knowledge/view/ad0e3b63).

File: packages/cockroachdb/changelog.yml (hunk 1 above)
Comment (ritalwar) [reply packages/cockroachdb/changelog.yml:3]: We're discussing offline about updating this changelog message for all packages, will adjust accordingly.

**Comment by lalit-satapathy** [comment]
> > @ritalwar, some packages are marked as: `version: "^8.12.0" `and some are marked as: `version: "^8.12.1"`, what is the difference here?
> 
> In version 8.12.0 of Kibana, the Fleet plugin was refactored, fixing how dynamic mappings for integrations are generated. However, this change caused mapping conflicts during data ingestion, impacting integrations like Prometheus and CockroachDB. The issue was resolved in version 8.12.1, therefore an upgrade to this version for these packages. Detailed information [here](https://support.elastic.dev/knowledge/view/ad0e3b63).

Ok, We are resolving the dynamic mapping issue in this PR as well? So it is only for prometheus type packages?

**Comment by ishleenk17** [comment]
> > > @ritalwar, some packages are marked as: `version: "^8.12.0" `and some are marked as: `version: "^8.12.1"`, what is the difference here?
> > 
> > 
> > In version 8.12.0 of Kibana, the Fleet plugin was refactored, fixing how dynamic mappings for integrations are generated. However, this change caused mapping conflicts during data ingestion, impacting integrations like Prometheus and CockroachDB. The issue was resolved in version 8.12.1, therefore an upgrade to this version for these packages. Detailed information [here](https://support.elastic.dev/knowledge/view/ad0e3b63).
> 
> Ok, We are resolving the dynamic mapping issue in this PR as well? So it is only for prometheus type packages?

Integrations using dynamic mappings would have this issue in 8.12

**Comment by lalit-satapathy** [comment]
> > > > @ritalwar, some packages are marked as: `version: "^8.12.0" `and some are marked as: `version: "^8.12.1"`, what is the difference here?
> > > 
> > > 
> > > In version 8.12.0 of Kibana, the Fleet plugin was refactored, fixing how dynamic mappings for integrations are generated. However, this change caused mapping conflicts during data ingestion, impacting integrations like Prometheus and CockroachDB. The issue was resolved in version 8.12.1, therefore an upgrade to this version for these packages. Detailed information [here](https://support.elastic.dev/knowledge/view/ad0e3b63).
> > 
> > 
> > Ok, We are resolving the dynamic mapping issue in this PR as well? So it is only for prometheus type packages?
> 
> Integrations using dynamic mappings would have this issue in 8.12

Should that be mixed with this work? This will disable secrets for someone running just 8.12.0.

**Comment by ritalwar** [comment]
> Ok, We are resolving the dynamic mapping issue in this PR as well? So it is only for prometheus type packages?

We're not resolving the dynamic mapping issue, it's already fixed in this [PR](https://github.com/elastic/kibana/pull/175970) and available from version 8.12.1. Due to system test failures in 8.12.0, we had to upgrade to 8.12.1.

**Comment by ishleenk17** [comment]
> > > > > @ritalwar, some packages are marked as: `version: "^8.12.0" `and some are marked as: `version: "^8.12.1"`, what is the difference here?
> > > > 
> > > > 
> > > > In version 8.12.0 of Kibana, the Fleet plugin was refactored, fixing how dynamic mappings for integrations are generated. However, this change caused mapping conflicts during data ingestion, impacting integrations like Prometheus and CockroachDB. The issue was resolved in version 8.12.1, therefore an upgrade to this version for these packages. Detailed information [here](https://support.elastic.dev/knowledge/view/ad0e3b63).
> > > 
> > > 
> > > Ok, We are resolving the dynamic mapping issue in this PR as well? So it is only for prometheus type packages?
> > 
> > 
> > Integrations using dynamic mappings would have this issue in 8.12
> 
> Should that be mixed with this work? This will disable secrets for someone running just 8.12.0.

The CI/CD would fail for these packages if the version is 8.12.
Either we have to force merge the packages, even though CI/CD fails for them? or resolve them with 8.12.1

**Review by agithomas — APPROVED** [review]
LGTM!

**Pull request merged**